### PR TITLE
feat(getCurrentStatus): add getCurrentStatus

### LINKED ISSUE #20: add `.status()` to inspect the status of agent
- sockets
- freeSockets
- requests
- createSocketCount
- closeSocketCount
- requestCount
- timeoutSocketCount

### PR DESCRIPTION
**Pull request by dead-horse**
closed #20

### CODE REVIEW

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
2.0.1

File: lib/agent.js [hunk 1]
```diff
@@ -83,6 +83,25 @@ Agent.prototype.createSocket = function (req, options) {
...
+    requestCount: this.requestCount,
+    freeSockets: inspect(this.freeSockets),
+    sockets: inspect(this.sockets),
+    requests: inspect(this.requests)
+  };
+};
+
+function inspect(obj) {
+  var res = {};
+  for (var key in obj) {
```
Comment (JacksonTian) [inline lib/agent.js]: Object.keys + for (var i = 0; i < length; i++) 会更快。

File: lib/agent.js [hunk 2]
```diff
@@ -83,6 +83,25 @@ Agent.prototype.createSocket = function (req, options) {
   return socket;
 };
 
+Agent.prototype.getCurrentStatus = function () {
```
Comment (JacksonTian) [inline lib/agent.js]: 叫`status()`更简洁些嘛

Comment (dead-horse) [inline lib/agent.js]: 太通用怕可能和以后原生的冲突

Comment (fengmk2) [inline lib/agent.js]: 对

**Pull request merged**